flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/operations/converters/SqlCreateOrAlterMaterializedTableConverter.java

```diff
@@ -0,0 +1,262 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.planner.operations.converters;
+
+import org.apache.flink.sql.parser.ddl.SqlCreateOrAlterMaterializedTable;
+import org.apache.flink.sql.parser.ddl.SqlTableColumn.SqlRegularColumn;
+import org.apache.flink.table.api.Schema;
+import org.apache.flink.table.api.ValidationException;
+import org.apache.flink.table.catalog.CatalogBaseTable.TableKind;
+import org.apache.flink.table.catalog.CatalogMaterializedTable;
+import org.apache.flink.table.catalog.CatalogMaterializedTable.LogicalRefreshMode;
+import org.apache.flink.table.catalog.CatalogMaterializedTable.RefreshMode;
+import org.apache.flink.table.catalog.CatalogMaterializedTable.RefreshStatus;
+import org.apache.flink.table.catalog.Column;
+import org.apache.flink.table.catalog.IntervalFreshness;
+import org.apache.flink.table.catalog.ObjectIdentifier;
+import org.apache.flink.table.catalog.ResolvedCatalogBaseTable;
+import org.apache.flink.table.catalog.ResolvedCatalogMaterializedTable;
+import org.apache.flink.table.catalog.ResolvedSchema;
+import org.apache.flink.table.catalog.TableChange;
+import org.apache.flink.table.catalog.TableChange.MaterializedTableChange;
+import org.apache.flink.table.catalog.TableDistribution;
+import org.apache.flink.table.operations.Operation;
+import org.apache.flink.table.operations.materializedtable.AlterMaterializedTableAsQueryOperation;
+import org.apache.flink.table.operations.materializedtable.CreateMaterializedTableOperation;
+import org.apache.flink.table.planner.operations.converters.table.MergeTableAsUtil;
+import org.apache.flink.table.planner.utils.MaterializedTableUtils;
+
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.SqlNodeList;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+/** A converter for {@link SqlCreateOrAlterMaterializedTable}. */
+public class SqlCreateOrAlterMaterializedTableConverter
+        extends AbstractCreateMaterializedTableConverter<SqlCreateOrAlterMaterializedTable> {
+
+    @Override
+    public Operation convertSqlNode(
+            SqlCreateOrAlterMaterializedTable sqlCreateOrAlterMaterializedTable,
+            ConvertContext context) {
+        final ObjectIdentifier identifier =
+                this.getIdentifier(sqlCreateOrAlterMaterializedTable, context);
+        return sqlCreateOrAlterMaterializedTable.isOrAlter() && tableExists(context, identifier)
+                ? handleAlter(sqlCreateOrAlterMaterializedTable, context)
+                : handleCreate(sqlCreateOrAlterMaterializedTable, context, identifier);
```

Comment:
can we instead just redirect call to either SqlCreateMaterializedTableConverter or SqlAlterMaterializedTableConverter?